bundle/Command/UpdateContentVisibilityCommand.php

```diff
@@ -0,0 +1,265 @@
+                $content = $this->repository->sudo(
+                    fn () => $this->repository->getContentService()->loadContent(
+                        $contentId,
+                        [$language->getLanguageCode()],
+                    ),
+                );
+            } catch (NotFoundException $exception) {
+                $this->logger->error(
+                    sprintf(
+                        'Content with id #%d does not exist: %s',
+                        $contentId,
+                        $exception->getMessage(),
+                    ),
+                );
+
+                continue;
+            }
+
+            try {
+                $action = $this->scheduledVisibilityService->updateVisibilityIfNeeded($content);
+            } catch (InvalidStateException $exception) {
+                $this->logger->error($exception->getMessage());
+
+                continue;
+            }
+
+            if ($action !== VisibilityUpdateResult::NoChange) {
+                $this->logger->info(
+                    sprintf(
+                        "Content '%s' with id #%d has been %s.",
+                        $content->getName(),
+                        $content->getId(),
+                        $action->value,
+                    ),
+                );
+            }
+
+            $this->style->progressAdvance();
+        }
+    }
+
+    private function getQueryBuilder(): QueryBuilder
+    {
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->select('id', 'initial_language_id')
+            ->from('ezcontentobject')
+            ->where('published != :unpublished')
+            ->orderBy('id', 'ASC')
+            ->setParameter('unpublished', 0);
+
+        return $query;
+    }
+
+    private function applyContentTypeLimit(QueryBuilder $query, array $contentTypeIds): void
+    {
+        $query->where(
+            $query->expr()->in('contentclass_id', ':content_type_ids'),
+        )->setParameter('content_type_ids', $contentTypeIds, Connection::PARAM_INT_ARRAY);
+    }
+
+    private function getPager(QueryBuilder $query, array $contentTypeIds): Pagerfanta
+    {
+        $countQueryBuilderModifier = static function (QueryBuilder $queryBuilder) use ($contentTypeIds): void {
+            $queryBuilder->select('COUNT(id) AS total_results')
+                ->from('ezcontentobject')
+                ->where('published != :unpublished')
+                ->setParameter('unpublished', 0)
+                ->setMaxResults(1);
+
+            if (count($contentTypeIds) > 0) {
+                $queryBuilder->where(
+                    $queryBuilder->expr()->in('contentclass_id', ':content_type_ids'),
+                )->setParameter('content_type_ids', $contentTypeIds, Connection::PARAM_INT_ARRAY);
+            }
+        };
+
+        return new Pagerfanta(new QueryAdapter($query, $countQueryBuilderModifier));
+    }
+
+    /**
+     * @throws NotFoundException
```

Comment:
Not thrown in this method.

Reply:
fixed in fa153b003652a49ea510b3b043c68e87d73d325f